### DISCUSSION TITLE
How can I display chinese characters with this liabrary?

### DISCUSSION THREAD

**Discussion by Gwiths** [Bodmer/TFT_eSPI#943] [Q&A - Fonts] [answered]
Hi, I had uploaded the data folder in which three vlw files(Unicode-Test-72.vlw, Final-Frontier-28.vlw, Latin-Hiragana-24.vlw) are included. No matter which one of them I unload with the function` tft.unloadFont(); `, chinese characters are replaced with some boxes. Is this library incompatible with chinese or just i used in a wrong way?
TIA.

**Comment by Bodmer** [comment, marked as answer]
The example font Unicode-Test-72.vlw only contains a few Chinese characters for test and demonstration purposes. The box drawn (often called Tofu!) indicates the character is not in the vlw file. You will need to generate your own vlw file containing the characters you need. There are 1000's of Chinese characters so you will need to select the ones you need to avoid creating a vlw file that is too big for memory. The vlw file can be created using the Processing IDE.

**Comment by Bodmer** [comment]
I wrote a Processing sketch to allow the font chanracters to be selected individually or as groups. You can[ find it here](https://github.com/Bodmer/TFT_eSPI/tree/master/Tools/Create_Smooth_Font/Create_font).

The Processing IDE is an ancester of the Arduino IDE and can be [downloaded here](https://processing.org/download/).

**Comment by darkbull** [comment]
> Hi, I had uploaded the data folder in which three vlw files(Unicode-Test-72.vlw, Final-Frontier-28.vlw, Latin-Hiragana-24.vlw) are included. No matter which one of them I unload with the function`tft.unloadFont();`, chinese characters are replaced with some boxes. Is this library incompatible with chinese or just i used in a wrong way?
> TIA.

我的fork里，加了些脚本生产中文，有对中文字体的生成及说明。如果文字数量多，会严重拖慢渲染。可以试一下u8g的gb2312，这个效果很好，不过只支持有限的文字大小。
